src/Task.gren

```diff
@@ -219,8 +192,21 @@ sequence fails.
 
 -}
 sequence : Array (Task x a) -> Task x (Array a)
-sequence tasks =
-    Array.foldr (map2 Array.pushFirst) (succeed []) tasks
+sequence =
+    Array.foldr (\task combined -> task |> andThen (\x -> map (Array.pushFirst x) combined)) (succeed []) 
+
+
+{-| Start with an array of tasks, and turn them into a single task that returns a
+array. The tasks will be run concurrently and if any task fails the whole array fails.
+
+    concurrent [ succeed 1, succeed 2 ] == succeed [ 1, 2 ]
+
+Additionally if any task fails, any tasks already in-flight will be cleaned up (e.g. an in-flgiht http request will be cancelled).
```

Comment:
I wonder if we should add a comment about why you'd ever choose `sequence` over `concurrent`? Like "If you're doing something really expensive, like making 100 http requests, you might want to do that sequentially instead of concurrently to avoid DDOSing the server" or something.

Reply:
I like that.

Do you think it's worth adding a note here about `concurrent` vs `parallel`? Because JS is single threaded, `concurrent` won't speed up CPU bound tasks, but IO bound tasks like http may see a speed boost.

Would need some friendlier wording and examples but something along those lines.

Reply:
Yeah, I think that would be good.